src/main/java/core/basesyntax/Application.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax;
+
+public class Application {
+    public static void main(String[] args) {
+        int figureNumber = 10;
+
+        int figuresNumberInSetOne = 0;
+
+        if (figureNumber % 2 == 0) {
+            figuresNumberInSetOne = figureNumber / 2;
+        } else {
+            figuresNumberInSetOne = figureNumber / 2 + 1;
+        }
+
+        Figure [] result = new Figure[figureNumber];
+
+        for (int i = 0; i < figuresNumberInSetOne; i++) {
+            FigureSupplier figureSupplier = new FigureSupplier();
+            result [i] = figureSupplier.getRandomFigure();
+            result [i].draw();
+        }
+
+        for (int i = figuresNumberInSetOne; i < figureNumber; i++) {
+            FigureSupplier figureSupplier = new FigureSupplier();

```

Comment:
As mentioned before, avoid creating a new instance of `FigureSupplier` inside the loop. Instead, create a single instance before starting the loop and use it to generate all the figures.